var/spack/repos/spack_repo/builtin/packages/py_pygraphviz/package.py

```diff
@@ -15,10 +15,32 @@ class PyPygraphviz(PythonPackage):
 
     license("BSD-3-Clause")
 
+    version("1.14", sha256="c10df02377f4e39b00ae17c862f4ee7e5767317f1c6b2dfd04cea6acc7fc2bea")
+    version("1.13", sha256="6ad8aa2f26768830a5a1cfc8a14f022d13df170a8f6fdfd68fd1aa1267000964")
+    version("1.12", sha256="8b0b9207954012f3b670e53b8f8f448a28d12bdbbcf69249313bd8dbe680152f")
+    version("1.11", sha256="a97eb5ced266f45053ebb1f2c6c6d29091690503e3a5c14be7f908b37b06f2d4")
     version("1.10", sha256="457e093a888128903251a266a8cc16b4ba93f3f6334b3ebfed92c7471a74d867")
     version("1.7", sha256="a7bec6609f37cf1e64898c59f075afd659106cf9356c5f387cecaa2e0cdb2304")
 
     depends_on("python@3.7:", type=("build", "run"))
     depends_on("python@3.8:", when="@1.10:", type=("build", "run"))
+    depends_on("python@3.10:", when="@1.12:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
     depends_on("graphviz")
+
+    def url_for_version(self, version):
+        url = "https://files.pythonhosted.org/packages/source/p/pygraphviz/pygraphviz-{}.{}"
+        if version >= Version("1.12"):
+            fext = "tar.gz"
+        else:
+            fext = "zip"
+        return url.format(version, fext)
+
+    # some how graphviz isn't detected during install.
+    @run_before("install")
+    def fix_setup(self):
```

Comment:
This should probably just be a [patch function](https://spack.readthedocs.io/en/latest/packaging_guide.html#patch-functions). Also to avoid issues with `lib` vs `lib64`, I would change it to this (untested):
```
lib_dir = self["graphviz"].libs.directories[0]
```

Reply:
This actually didn't work. it seems '..libs.directories[0]' yields just a '/' :thinking:

Reply:
I think the problem is that the libraries in graphviz package aren't named 'libgraphviz...' but like `libgv..` etc
so, I tried defining a custom libs property in graphviz, but still spack can't find the libraries
```
       @property
        def libs(self):
            libraries = ["libgvc", "libcdt", "libcgraph", "libgvpr", "libpathplan", "libxdot"]
            return find_libraries(libraries, root=self.prefix, recursive=True)
```

Reply:
actually. it's now fixed. I guess the assumption is that patch is applied in build directory hence no need to specify it.